docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+1. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA.
+:exclamation: Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. **Verify the setup**:
+   1. Run the `seedu.trippie.Trippie` and try a few commands.
+   2. [Run the tests](TestingGuide.md) to ensure they all pass. 
+
+## 3.0 Design
+
+This section explains the high-level design of the application. Given below is an overview of each component and a more detailed explanation of the architecture. Trippie is the main class which manages the initialization of the relevant classes and their execution.
+
+
+### 3.1 Architecture (Felix)
+
+<kbd>
+
+![UML Diagram Main](https://i.imgur.com/V3vUpoo.jpg)
+<center><i>Figure 1: Overall Architecture</i></center>
+
+</kbd>
+
+The Architecture Diagram given above explains the high-level design of Trippie. Shown below is a diagram of Trippie's high-level structure.
+
+The main class of`Trippie` is called [`Trippie.java`](https://github.com/AY2021S1-CS2113T-W11-2/tp/blob/master/src/main/java/seedu/trippie/Trippie.java?) It is responsible for:
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of three components.
+- **UI** : The UI of the App.
+- **Parser** : Parses user input and its information to respective commands.
+- **Storage** : Reads data from, and write data to, the text file.
+- **Command** : Executes each command respectively.
+
+The Sequence diagram below shows how the program flows across the different classes when parsing or executing user commands.
+
+![UML Sequence Diagram](https://i.imgur.com/hHq7ltY.png)
+<center><i>Figure 2: Sequence Diagram during Command execution</i></center>
+
+### 3.2 Ui (Felix)
+
+The UI class is in charge of the user's input, from the `readCommand` or `getLine` methods. Other functions of the Ui is to provide shorthands of Trippie's outputs, such as `greetUser` or `showFarewells`.
+
+### 3.3 Parser (Wei Shuang)
+The `Parser` class is implemented to parse the user's input, and returns a `Command` object. The `Trippie` class will read in this object and call the `execute` method under the object.
+
+When the `Parser` class is unable to successfully parse the user's input, `TrippieIllegalCommandException` will be thrown. (Refer to 3.7 Exceptions)
+
+### 3.4 TrippieData (Ivander)
+
+The `TrippieData` class is implemented to store all of your data during Trippie's runtime. Below is an architecture diagram of the class.
+
+<kbd>
+
+![TrippieData UML Diagram](https://i.imgur.com/3eAMiob.jpg)
```

Comment:
Should the Currency be connected with an arrow instead of as a composition?
<img width="306" alt="image" src="https://user-images.githubusercontent.com/51308797/97525494-e5487880-19e1-11eb-8e33-7deaca20caa5.png">